pkg/aws/ec2.go

```diff
@@ -306,16 +306,20 @@ func AssertEC2InstancesBalancedInSubnets(t *testing.T, ctx context.Context, inpu
 	for _, subnet := range input.Subnets {
 		subnetIDToInstanceCountMap[*subnet.SubnetId] = 0
 	}
+
 	for _, instance := range input.Instances {
 		if _, ok := subnetIDToInstanceCountMap[*instance.SubnetId]; !ok {
-			// Fail the test if we get a subnet from an instance that was not in the original list of subnets.
-			t.Fatalf("Subnet %s associated with instance %s is not in the input list of subnets.",
+			t.Logf("Subnet %s associated with instance %s is not in the input list of subnets.",
 				*instance.SubnetId, *instance.InstanceId)
+			subnetIDToInstanceCountMap[*instance.SubnetId] = 0
```

Comment:
This is something I'm looking for opinions on... what should be the behavior when an EC2 instance is passed in with a subnet ID that was not in the input subnet list?

Reply:
Agreed, I think it makes more sense to look at the list of subnets that the instances are placed in. It'll also make the code a little simpler. Will push a change soon!